src/modules/dashboards/saga.ts

```diff
@@ -343,6 +349,69 @@ export function* persistDashboardsOrder({
   }
 }
 
+export function* cloneDashboard({
+  payload,
+}: ReturnType<typeof cloneDashboardAction>) {
+  const { dashboardId } = payload;
+
+  const blobApi = yield getContext(BLOB_API);
+
+  const model = yield blobApi.getDashboardById(dashboardId);
+
+  const uniqueIdWidgets = model.widgets.map((widget) => ({
+    ...widget,
+    id: createWidgetId(),
+  }));
+
+  const newDashboardId = uuid();
+  const metaData = yield blobApi.getDashboardMetadataById(dashboardId);
+  const newMetaData = {
+    ...metaData,
+    id: newDashboardId,
+    title: metaData.title ? `${metaData.title} Clone` : 'Clone',
+    isPublic: false,
+    lastModificationDate: +new Date(),
+  };
+
+  const newModel = {
+    ...model,
+    widgets: uniqueIdWidgets,
+  };
+
+  try {
+    yield blobApi.saveDashboard(newDashboardId, newModel, newMetaData);
+
+    yield put(addClonedDashboard(newMetaData));
+
+    const notificationManager = yield getContext(NOTIFICATION_MANAGER);
+    yield notificationManager.showNotification({
+      type: 'info',
+      message: 'notifications.dashboard_cloned',
+      autoDismiss: true,
+    });
+
+    const state: RootState = yield select();
+    const activeDashboard = getActiveDashboard(state);
+
+    if (activeDashboard) {
+      yield put(setActiveDashboard(newDashboardId));
```

Comment:
For better user experience - I believe it will be better to change route after dashboard is registered and initialized.